File: dotnet-desktop-guide/winforms/migration/snippets/clipboard-dataobject-net10/net/vb/SetDataAsJsonExamples.vb
Imports System
Imports System.Windows.Forms

Namespace ClipboardExamples
    Public Class SetDataAsJsonExamples
        Public Class Person
            Public Property Name As String
            Public Property Age As Integer
        End Class

        Public Class AppSettings
            Public Property Theme As String
            Public Property AutoSave As Boolean
        End Class

        ' <AutomaticFormatInference>
        Public Shared Sub AutomaticFormatInferenceExample()
            Dim person As New Person With {.Name = "Alice", .Age = 25}

            ' Use the type name as the format
            Clipboard.SetDataAsJson(GetType(Person).FullName, person)  ' Uses "Person" as the format

            ' Retrieve the data and infer the format automatically
            Dim retrievedPerson As Person = Nothing
            If Clipboard.TryGetData(retrievedPerson) Then
                Console.WriteLine($"Retrieved: {retrievedPerson.Name}")
            End If
        End Sub
        ' </AutomaticFormatInference>

        ' <CustomFormat>
        Public Shared Sub CustomFormatExample()
            Dim settings As New AppSettings With {.Theme = "Dark", .AutoSave = True}

            ' Use a custom format for better organization
            Clipboard.SetDataAsJson("MyApp.Settings", settings)
        End Sub
        ' </CustomFormat>
    End Class
End Namespace

File: dotnet-desktop-guide/winforms/migration/snippets/clipboard-dataobject-net10/net/csharp/SetDataAsJsonExamples.cs
using System;
using System.Windows.Forms;

namespace ClipboardExamples
{
    public class SetDataAsJsonExamples
    {
        public class Person
        {
            public string Name { get; set; }
            public int Age { get; set; }
        }

        public class AppSettings
        {
            public string Theme { get; set; }
            public bool AutoSave { get; set; }
        }

        // <AutomaticFormatInference>
        public static void AutomaticFormatInferenceExample()
        {
            var person = new Person { Name = "Alice", Age = 25 };

            // Use the type name as the format
            Clipboard.SetDataAsJson(typeof(Person).FullName, person);

            // Retrieve the data and infer the format automatically
            if (Clipboard.TryGetData(out Person retrievedPerson))
            {
                Console.WriteLine($"Retrieved: {retrievedPerson.Name}");
            }
        }
        // </AutomaticFormatInference>

        // <CustomFormat>
        public static void CustomFormatExample()
        {
            var settings = new AppSettings { Theme = "Dark", AutoSave = true };

            // Use a custom format for better organization
            Clipboard.SetDataAsJson("MyApp.Settings", settings);
        }
        // </CustomFormat>
    }
}
